lexer fails to build on macos-11+

When doing make on phoebe1, the following error is displayed on macos 12:

Making all in libphoebe
test -f phoebe_constraints.grm.c || /bin/sh ../autoconfig/ylwrap phoebe_constraints.grm.y y.tab.c phoebe_constraints.grm.c y.tab.h `echo phoebe_constraints.grm.c | sed -e s/cc$/hh/ -e s/cpp$/hpp/ -e s/cxx$/hxx/ -e s/c++$/h++/ -e s/c$/h/` y.output phoebe_constraints.grm.output -- bison -y -d -p pc 
/Library/Developer/CommandLineTools/usr/bin/make  all-am
test -f phoebe_constraints.grm.c || /bin/sh ../autoconfig/ylwrap phoebe_constraints.grm.y y.tab.c phoebe_constraints.grm.c y.tab.h `echo phoebe_constraints.grm.c | sed -e s/cc$/hh/ -e s/cpp$/hpp/ -e s/cxx$/hxx/ -e s/c++$/h++/ -e s/c$/h/` y.output phoebe_constraints.grm.output -- bison -y -d -p pc 
/bin/sh ../libtool  --tag=CC   --mode=compile gcc -DHAVE_CONFIG_H -I.    -Wall  -g -O2 -MT libphoebe_la-phoebe_constraints.grm.lo -MD -MP -MF .deps/libphoebe_la-phoebe_constraints.grm.Tpo -c -o libphoebe_la-phoebe_constraints.grm.lo `test -f 'phoebe_constraints.grm.c' || echo './'`phoebe_constraints.grm.c
libtool: compile:  gcc -DHAVE_CONFIG_H -I. -Wall -g -O2 -MT libphoebe_la-phoebe_constraints.grm.lo -MD -MP -MF .deps/libphoebe_la-phoebe_constraints.grm.Tpo -c phoebe_constraints.grm.c  -fno-common -DPIC -o .libs/libphoebe_la-phoebe_constraints.grm.o
phoebe_constraints.grm.c:1079:16: error: implicit declaration of function 'pclex' is invalid in C99 [-Werror,-Wimplicit-function-declaration]
      yychar = yylex ();
               ^
phoebe_constraints.grm.c:69:25: note: expanded from macro 'yylex'
#define yylex           pclex
                        ^
1 error generated.
make[2]: *** [libphoebe_la-phoebe_constraints.grm.lo] Error 1
make[1]: *** [all] Error 2
make: *** [all-recursive] Error 1

--- a/phoebe-lib/libphoebe/phoebe_constraints.grm.y
+++ b/phoebe-lib/libphoebe/phoebe_constraints.grm.y
@@ -11,6 +11,7 @@
 /*#include "phoebe_constraints.lng.h"*/
 #include "phoebe_parameters.h"
 
+extern int yylex(void);
 extern int yyerror (const char *str);
 extern int intern_constraint_add_to_table (PHOEBE_ast *ast);
 %}

